### PR TITLE
[#9536] Reduce E2E tests flakiness due to underlying test structure

### PR DESCRIPTION
**Pull request by wkurniawan07**
Part of #9536 

This aims to reduce the flakiness of E2E tests due to the underlying test structure.

- Remove `BrowserPool` usage. Previously, as UI tests are a part of developer's routine, it was thought that this can save some time/memory resource as starting up browser takes a long time. Since then: (1) Selenium has evolved such that the time is not that long, (2) E2E tests are made optional and only to be run sparingly, (3) our need to have fresh browser for each test has been increasing. All of them warrant to stop the usage of `BrowserPool`.
  - Add a flag to indicate whether the browser is to be closed after failed test run. Useful for local debugging. Successful test run will always close the browser as there is no point in keeping it open.
- Run each test in incognito to ensure fresh starting state.
- (Important!) Disable Angular service worker for the app used in E2E test. It turns out that this is a major aggravating factor as failed Angular chunk loading will be persisted if service worker is used.

Also:

- Timeout is reduced to 10 (for page loads) as this is sufficient. ~There are many runs that show that even 5 is sufficient, but 10 is the safer number.~ For other purposes, 5 is already sufficient.
- Apply `afterTestClosure` after every run. Previously this was not done due to God mode (old term for snapshot testing) being part of UI tests, whereby failed HTML comparison will easily fill up the entire logs if repeated 3-4 times. Now that snapshot testing has been moved out, the `afterTestClosure` is safe to apply after every test run and can capture different causes of intermittent failures in different runs.
- Add one test retry as a last safety net.

### CODE REVIEW

**Review by jtankw3 — APPROVED** [review]
LGTM! Is this still ongoing?

**Comment by wkurniawan07** [comment]
@jtankw3 Yes, still ongoing. I'm still experimenting with 1-2 more improvements, the ones that I included so far are the ones I'm already confident with.

File: .travis.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
         - mv src/e2e/resources/test.travis.properties src/e2e/resources/test.properties
         - ./gradlew createConfigs testClasses generateTypes
         - npm ci
-        - npm run build -- --progress=false
+        - npm run build -- --progress=false --serviceWorker=false
```
Comment (jtankw3) [inline .travis.yml:80]: I believe this is by default false so I am not very sure if it is needed here?

**Review by jtankw3 — APPROVED** [review]
LGTM besides the one query.

File: .travis.yml (hunk 1 above)
Comment (wkurniawan07) [reply .travis.yml:80]: https://github.com/TEAMMATES/teammates/blob/4cce50f625ceb0e411d34168424bab7c546cca3a/angular.json#L106

**Pull request merged**